### ISSUE TITLE
question about natten library bug during train

### ISSUE THREAD

**Issue by EcustBoy** [jchengai/planTF#3]
Hi~ author, I follow your  installation and set up the conda env same with your suggest, but when run training, it will throw a bug here:
![natten bug](https://github.com/jchengai/planTF/assets/32817744/4e8cd95c-1457-4a6b-baef-f2c1187a9cda)

but when I add a breakpoint in line 50, stop for one second, this line can run normally, it seems like the natten library isn't stable enough in the multi-threads training mode by ray?

have you met such error before? I'll thanks for answer

**Comment by jchengai** [comment]
Hi, @EcustBoy, that's strange as I have not encountered similar errors. Can you try the following command and see if it works?

```
CUDA_VISIBLE_DEVICES=0 python run_training.py \      
  py_func=train +training=train_planTF \
  worker=single_machine_thread_pool worker.max_workers=32 \
  data_loader.params.batch_size=4 data_loader.params.num_workers=4 \
  lr=1e-3 epochs=25 warmup_epochs=3 weight_decay=0.0001 scenario_filter=mini
```

**Comment by being0606** [comment]
<img width="1066" height="331" alt="natten install error" src="https://github.com/user-attachments/assets/40a87840-7cd4-49d1-8cd7-212baaab60a6" />

# Troubleshooting “No module named ‘natten’” on PyTorch 2.7.1 + CUDA 12.6

I ran into the same import error even after installing the CUDA-enabled wheel. Here’s the step-by-step I followed to diagnose and fix it:

---


```bash
# Python version (must be ≥ 3.9)
python --version
# PyTorch version
python -c "import torch; print('PyTorch:', torch.__version__)"
# CUDA version compiled with PyTorch
python -c "import torch; print('CUDA:', torch.version.cuda)"
```

in my case, I solved it.
```bash
pip install natten==0.17.5+torch270cu126 \
  -f https://whl.natten.org
```
